compiler/cli/src/org/jetbrains/kotlin/cli/js/K2JSCompiler.java

```diff
@@ -107,29 +106,38 @@ protected ExitCode doExecute(
         Project project = environmentForJS.getProject();
         List<JetFile> sourcesFiles = environmentForJS.getSourceFiles();
 
-        if (arguments.verbose) {
-            reportCompiledSourcesList(messageCollector, sourcesFiles);
+        if (arguments.outputFile == null) {
+            messageSeverityCollector.report(CompilerMessageSeverity.ERROR, "Specify output file via -output", CompilerMessageLocation.NO_LOCATION);
+            return ExitCode.INTERNAL_ERROR;
         }
 
-        if (arguments.outputFile == null) {
-            messageCollector.report(CompilerMessageSeverity.ERROR, "Specify output file via -output", CompilerMessageLocation.NO_LOCATION);
+        if (messageSeverityCollector.anyReported(CompilerMessageSeverity.ERROR)) {
             return ExitCode.INTERNAL_ERROR;
         }
 
+        if (sourcesFiles.isEmpty()) {
+            messageSeverityCollector.report(CompilerMessageSeverity.ERROR, "No source files", CompilerMessageLocation.NO_LOCATION);
+            return COMPILATION_ERROR;
```

Comment:
It seems in `K2JVMCompiler.doExecute' `INTERNAL_ERROR` is used only for return from `catch` block. For `K2JSCompiler.doExecute' we have sometimes `INTERNAL_ERROR` and sometimes `COMPILATION_ERROR`.